design/comment-scanning.md

```diff
@@ -0,0 +1,110 @@
+# Polymer Analyzer - Comment Directives & Comment Annotations
+> Author @FredKSchott
+> Status: Draft
+> Last Update: 2017-04-25
+
+## Objective
+- To create a paradigm for formatting document metadata and instructions for analysis.
+- Immediately driving this discussion is the need to support polymer-linter directives which can enable/disable lint rules within entire documents and/or single sections of documents.
+
+## Goals
+- Allow the analyzer to surface the information contained within polymer-lint directives so that the linter can properly report & ignore warnings.
+- Create a more general definition for a directive that is easy-to-use for code authors and fits into the data model of the analyzer.
+- Create a more general definition for metadata (ex: @polymerElement) that is attached to an already-existing feature.
```

Comment:
I don't think jsdoc tags should be a goal here, but I do think that attached directive should be. We need a way to select rules for an element, and a range is more cumbersome than an attached comment.